e2e/cypress/integration/enterprise/system_console/authentication/authentication3_spec.js

```diff
@@ -0,0 +1,285 @@
+// ***************************************************************
+
+// Group: @system_console @authentication
+
+import * as TIMEOUTS from '../../../../fixtures/timeouts';
+
+import {getRandomId} from '../../../../utils';
+
+describe('Authentication Part 3', () => {
+    let testUser;
+
+    before(() => {
+        cy.apiRequireLicense();
+
+        cy.apiInitSetup().then(() => {
+            cy.apiCreateUser().then(({user: newUser}) => {
+                testUser = newUser;
+            });
+        });
+    });
+
+    beforeEach(() => {
+        // # Log in as a admin.
+        cy.apiAdminLogin({failOnStatusCode: false});
+    });
+
+    after(() => {
+        // # Logout as any current user
+        cy.apiLogout();
+
+        // # Log in as a admin.
+        cy.apiAdminLogin({failOnStatusCode: false});
+
+        cy.apiUpdateConfig({
+            EmailSettings: {
+                EnableSignInWithEmail: true,
+                EnableSignInWithUsername: true,
+            },
+        });
+    });
+
+    it('MM-T1767 - Email signin false Username signin true', () => {
+        cy.apiUpdateConfig({
+            EmailSettings: {
+                EnableSignInWithEmail: false,
+                EnableSignInWithUsername: true,
+            },
+            LdapSettings: {
+                Enable: false,
+            },
+        });
+
+        cy.apiLogout();
+
+        // # Go to front page
+        cy.visit('/login');
+
+        // * Make sure input section for username contains username and not email
+        cy.findByTestId('loginId').invoke('attr', 'placeholder').should('contain', 'Username').and('not.contain', 'Email');
+    });
+
+    it('MM-T1768 - Email signin true Username signin true', () => {
+        cy.apiUpdateConfig({
+            EmailSettings: {
+                EnableSignInWithEmail: true,
+                EnableSignInWithUsername: true,
+            },
+            LdapSettings: {
+                Enable: false,
+            },
+        });
+
+        cy.apiLogout();
+
+        // # Go to front page
+        cy.visit('/login');
+
+        // * Make sure input section for username contains username and not email
+        cy.findByTestId('loginId', {timeout: TIMEOUTS.ONE_MIN}).invoke('attr', 'placeholder').should('contain', 'Username').and('contain', 'Email');
+    });
+
+    it('MM-T1769 - Email signin true Username signin false', () => {
+        cy.apiUpdateConfig({
+            EmailSettings: {
+                EnableSignInWithEmail: true,
+                EnableSignInWithUsername: false,
+            },
+            LdapSettings: {
+                Enable: false,
+            },
+        });
+
+        cy.apiLogout();
+
+        // # Go to front page
+        cy.visit('/login');
+
+        // * Make sure input section for username contains username and not email
+        cy.findByTestId('loginId', {timeout: TIMEOUTS.ONE_MIN}).invoke('attr', 'placeholder').should('contain', 'Email').and('not.contain', 'Username');
+    });
```

Comment:
All three of the true/true, true/false, and false/true tests could be written as one test with less duplicated code.

Reply:
Yah you're right. I didn't realize it. However, I kept them separate for that reason however I think it's better to combine and reduce code. I've done that too.